### PR TITLE
[RFC] Less global magic

### PR DESCRIPTION
**Pull request by bb010g**
This will make every previous global variable local and properly `require`d. ~~Currently in progress, but `administration` and the other changed plugins should give you a good feel for how it it is in use in the wild.~~ Also fixes a bunch of triggers, makes a future switch to regex easier, and fixes some leakage of HTTP global state from `lastfm`.

---
This change is [Reviewable](https://reviewable.io/reviews/topkecleon/otouto/48)

### CODE REVIEW

File: plugins/about.lua [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
 
-local action = function(msg)
+function about:action(msg)
 
 	-- Filthy hack, but here is where we'll stop forwarded messages from hitting
 	-- other plugins.
 	if msg.forward_from then return end
 
-	local message = config.about_text .. '\nBased on @otouto v'..version..' by topkecleon.'
+	local message = self.config.about_text .. '\nBased on @otouto v'..self.version..' by topkecleon.'
```
Comment (bb010g) [inline plugins/about.lua]: `self.version` will be fixed next commit.

**Comment by topkecleon** [comment]
Does it run?

**Comment by bb010g** [comment]
Ish. I'm working on finishing up the rest of the plugins (couldn't really work on it yesterday due to computer hardware issues), but the first couple should work, and the core should work. I was mainly putting this out here now so you could give feedback if you wanted.

**Comment by bb010g** [comment]
Ok. This breaks pretty much every plugin not in `otouto` proper, but the refactoring's done as far as I know right now. Test at your own risk.

**Comment by bb010g** [comment]
Review status: 0 of 54 files reviewed at latest revision, 6 unresolved discussions.

---

*[bindings.lua, line 11 \[r2\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF4sHQ8OKtr9T8aMS_8:-KF4sHQCydui5UbUpaaO:1945640469) ([raw file](https://github.com/topkecleon/otouto/blob/550d0743b7170a4a067f3fa546a43857524dbc51/bindings.lua#L11)):*
Right now this is gone and the bot will just fail saying that it can't find `self.BASE_URL`. Do you want this back in `bot.lua` when it starts up, or are you fine with it being gone?

---

*[utilities.lua, line 277 \[r3\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF4tbUbKCwqhewkKdiD:-KF4tbUbKCwqhewkKdiE:-517838757) ([raw file](https://github.com/topkecleon/otouto/blob/0e8a2693fbcba74d5696e1cb28e2d87b2776e13a/utilities.lua#L277)):*
This needs a better name, but I couldn't think of one. If you have any suggestions, please give them.

---

*[plugins/about.lua, line 18 \[r1\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF2X5nr6MDwax5wXOnp-r1-18:-KF4ryzhwt3ZbZBmP-CT:-499146009) ([raw file](https://github.com/topkecleon/otouto/blob/acb679f8fa78349415de4ed070322afcaaf2cac4/plugins/about.lua#L18)):*
Done.

---

*[plugins/administration.lua, line 1135 \[r3\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF4uXkXPlXqWTlk9oUt:-KF4uXkXPlXqWTlk9oUu:-2125148043) ([raw file](https://github.com/topkecleon/otouto/blob/0e8a2693fbcba74d5696e1cb28e2d87b2776e13a/plugins/administration.lua#L1135)):*
`self.admin_temp`

---

*[plugins/cats.lua, line 17 \[r3\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF4usA1iSL-1ZE1PsKh:-KF4usA1iSL-1ZE1PsKi:-414552169) ([raw file](https://github.com/topkecleon/otouto/blob/0e8a2693fbcba74d5696e1cb28e2d87b2776e13a/plugins/cats.lua#L17)):*
`triggers(self.info.username)`, move to `init`

---

*[plugins/gImages.lua, line 2 \[r3\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF4vUp5kO6TGZs8FqeP:-KF4vUp5kO6TGZs8FqeQ:128633311) ([raw file](https://github.com/topkecleon/otouto/blob/0e8a2693fbcba74d5696e1cb28e2d87b2776e13a/plugins/gImages.lua#L2)):*
Spelling

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/topkecleon/otouto/48)*

**Comment by bb010g** [comment]
Review status: 0 of 54 files reviewed at latest revision, 4 unresolved discussions.

---

*[bindings.lua, line 11 \[r2\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KF4sHQ8OKtr9T8aMS_8:-KF7al2jEip4kr7KLJCs:2122704916) ([raw file](https://github.com/topkecleon/otouto/blob/550d0743b7170a4a067f3fa546a43857524dbc51/bindings.lua#L11)):*
Nevermind, this was already moved! Good job, past me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/topkecleon/otouto/48)*

**Comment by bb010g** [comment]
Reviewed 4 of 41 files at r1, 36 of 49 files at r2, 2 of 4 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 2 of 2 files at r11.
Review status: 49 of 54 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/topkecleon/otouto/48)*

**Comment by topkecleon** [comment]
Sorry to break your automatic merging potential. :)

**Comment by bb010g** [comment]
Reviewed 2 of 6 files at r12.
Review status: 49 of 54 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/topkecleon/otouto/48)*

**Comment by bb010g** [comment]
Reviewed 2 of 6 files at r12.
Review status: 51 of 54 files reviewed at latest revision, 6 unresolved discussions.

---

*[bot.lua, line 30 \[r12\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KFFqSDTWRhQ8MR9Gd-M:-KFFqSDTWRhQ8MR9Gd-N:-896532776) ([raw file](https://github.com/topkecleon/otouto/blob/80930f389c8126d9d18cbbf9e90947e6a55ecdc5/bot.lua#L30)):*
Should be `plugins.whatever` for loading consistency. This goes along with removing the `.lua`s from `plugins.lua`.

---

*[plugins/administration.lua, line 327 \[r12\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KFFte1kfCI7t3DPBz2f:-KFFte1kfCI7t3DPBz2g:-1783573639) ([raw file](https://github.com/topkecleon/otouto/blob/80930f389c8126d9d18cbbf9e90947e6a55ecdc5/plugins/administration.lua#L327)):*
`self`

---

*[plugins/administration.lua, line 375 \[r12\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KFFsSsvYV8kVZdhnHJd:-KFFsSsvYV8kVZdhnHJe:-1783573639) ([raw file](https://github.com/topkecleon/otouto/blob/80930f389c8126d9d18cbbf9e90947e6a55ecdc5/plugins/administration.lua#L375)):*
`self`

---

*[plugins/administration.lua, line 802 \[r12\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KFFtO3I-gY5FeLwaiye:-KFFtO3I-gY5FeLwaiyf:-1783573639) ([raw file](https://github.com/topkecleon/otouto/blob/80930f389c8126d9d18cbbf9e90947e6a55ecdc5/plugins/administration.lua#L802)):*
`self`

---

*[plugins/administration.lua, line 990 \[r12\]](https://reviewable.io:443/reviews/topkecleon/otouto/48#-KFFtZDNXmL-AhZhaJuQ:-KFFtZDOPNdmwvIkF5yt:-1783573639) ([raw file](https://github.com/topkecleon/otouto/blob/80930f389c8126d9d18cbbf9e90947e6a55ecdc5/plugins/administration.lua#L990)):*
`self`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/topkecleon/otouto/48)*

**Comment by bb010g** [comment]
Reviewed 1 of 49 files at r2, 5 of 5 files at r13, 1 of 1 files at r14.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/topkecleon/otouto/48)*

**Comment by bb010g** [comment]
Ok. With that, the comprehensive review (thanks to @TiagoDanin for catching a missing `return utilities`) done, this should be ready to merge after some testing just to make sure. Only thing I don't quite like is `utilities.INVOCATION_PATTERN`, but I can't come up with a better name. Anyhow, that can be solved later, and it's mostly unnoticed thanks to `utilities.triggers()`.

**Comment by topkecleon** [comment]
Will commence private testing.

**Comment by christoga** [comment]
Fix the git conflict dude

**Pull request merged**